src/components/scenes/CoinRankingScene.tsx

```diff
@@ -0,0 +1,290 @@
+      console.error(msg)
+      showError(msg)
+      return
+    }
+    const newIndex = index + 1 >= percentChangeOrder.length ? 0 : index + 1
+    const newTimeFrame = percentChangeOrder[newIndex]
+    setPercentChangeTimeFrame(newTimeFrame)
+  })
+
+  const handlePriceSubText = useHandler(() => {
+    const newPriceSubText = priceSubText === 'marketCap' ? 'volume24h' : 'marketCap'
+    setPriceSubText(newPriceSubText)
+  })
+
+  const handleOnChangeText = useHandler((input: string) => {
+    setSearchText(input)
+  })
+  const handleTextFieldFocus = useHandler(() => {
+    setSearching(true)
+  })
+  const handleSearchDone = useHandler(() => {
+    setSearchText('')
+    setSearching(false)
+    textInput.current?.blur()
+  })
+  const handleSubmit = useHandler(() => {
+    if (searchText === '') {
+      setSearching(false)
+    }
+  })
+
+  React.useEffect(() => {
+    return () => {
+      if (timeoutHandler.current != null) {
+        clearTimeout(timeoutHandler.current)
+      }
+      mounted.current = false
+    }
+  }, [])
+
+  useAsyncEffect(async () => {
+    const queryLoop = async () => {
+      try {
+        let start = 1
+        debugLog(LOG_COINRANK, `queryLoop dataSize=${dataSize} requestDataSize=${requestDataSize}`)
+        while (start < requestDataSize) {
+          const url = `v2/coinrank?fiatCode=${fiatCode}&start=${start}&length=${QUERY_PAGE_SIZE}`
+          const response = await fetchRates(url)
+          if (!response.ok) {
+            const text = await response.text()
+            console.warn(text)
+            break
+          }
+          const replyJson = await response.json()
+          const listings = asCoinrankings(replyJson)
+          for (let i = 0; i < listings.data.length; i++) {
+            const rankIndex = start - 1 + i
+            const row = listings.data[i]
+            coinRankings[rankIndex] = row
+            debugLog(LOG_COINRANK, `queryLoop: ${rankIndex.toString()} ${row.rank} ${row.currencyCode}`)
+          }
+          start += QUERY_PAGE_SIZE
+        }
+        setDataSize(coinRankings.length)
+      } catch (e: any) {
+        console.warn(e.message)
+      }
+      timeoutHandler.current = setTimeout(queryLoop, LISTINGS_REFRESH_INTERVAL)
+    }
+    if (timeoutHandler.current != null) {
+      clearTimeout(timeoutHandler.current)
+    }
+    queryLoop().catch(e => debugLog(LOG_COINRANK, e.message))
+  }, [requestDataSize])
+
+  const listdata: number[] = React.useMemo(() => {
+    debugLog(LOG_COINRANK, `Updating listdata dataSize=${dataSize} searchText=${searchText}`)
+    const out = []
+    for (let i = 0; i < dataSize; i++) {
+      const cr = coinRankings[i]
+      if (searchText === '') {
+        out.push(i)
+      } else {
+        if (cr.currencyCode.toLowerCase().includes(searchText.toLowerCase()) || cr.currencyName.toLowerCase().includes(searchText.toLowerCase())) {
+          out.push(i)
+        }
+      }
+    }
+    return out
+  }, [dataSize, searchText])
```

Comment:
Address exhaustive dep warning

```suggestion
    }, [coinRankings, dataSize, searchText])
```